core/src/core/pmmr.rs

```diff
@@ -371,10 +374,12 @@ where
 		// creation of another parent.
 		while bintree_postorder_height(pos + 1) > height {
 			let left_sibling = bintree_jump_left_sibling(pos);
-			let left_elem = self.backend
-				.get(left_sibling, false)
-				.expect("missing left sibling in tree, should not have been pruned");
-			current_hash = left_elem.0 + current_hash;
+
+			let left_hash = self.backend
+				.get_from_file(left_sibling)
```

Comment:
We need to bypass the rm_log here and go straight to the underlying file.
We may have "removed" the left_sibling but we still have access to the hash in the file.

Reply:
That may be the same as my misunderstanding with the peaks, I don't see what tree construction would lead to this. Maybe we're missing a test in `prune` instead and overpruning?